pkg/util/util.go

```diff
@@ -146,3 +147,16 @@ func ContainsWholeWord(source string, search string) bool {
 	}
 	return false
 }
+
+func CheckPortAvailable(addr string, port string) bool {
```

Comment:
let's try to come up with a solution that doesn't require port scanning

Reply:
Is your concern around if this would make harvest start slow? Pls suggest any alternative.